engine/src/desktop-menu.cpp

```diff
@@ -444,7 +444,8 @@ void MCButton::macopenmenu(void)
                 // SN-2014-08-25: [[ Bug 13240 ]] We need to keep the actual popup_menustring,
                 //  in case some menus are nested
                 MCStringRef t_menupick;
-                t_menupick = s_popup_menupick;
+                // PM-2015-02-09: [[ Bug 14521 ]] Make sure we don't pass nil to message_with_valueref_args
+                t_menupick = !MCStringIsEmpty(s_popup_menupick) ? s_popup_menupick : kMCEmptyString;
```

Comment:
I'm not clear on the recipe - do you mean the menu selection is cancelled?

If it is cancelled then I'm not sure it should be sending a menupick message at all...